### PR TITLE
Copter: RC override arming check fix.

### LINKED ISSUE #21921: Copter: RC not found pre-arm if only RC_OVERRIDES are used
Testing latest Cotper-4.4.0-dev I've found that if only a joystick is used (from MP, aka RC_OVERRIDES) then a pre-arm failure appears, "PreArm: RC not found".  I can see the RC values changing in MP's Setup >> RC Calibration screen and RC failsafes are triggered or cleared as the joystick is enabled or disabled.

image

image

### PR DESCRIPTION
**Pull request by IamPete1**
Fixes #21921

Not tested.

### CODE REVIEW

**Comment by rmackay9** [comment]
very nice, I will test it!

**Comment by rmackay9** [comment]
I've successfully tested this.  thanks!

In my test I first left the joystick unplugged and the warning appeared.  I then plugged in the joystick (and enabled it) and the warning disappeared.  In the screen shot below we can see the original warning.. then higher up that it's not being displayed along with the other pre-arm failures.
image

**Review by rmackay9 — APPROVED** [review]
This looks good to me.  I didn't test with a regular RC receiver to see that the original desired behaviour persists but I'm quite sure it will work.

Maybe @muramura could give this a test to confirm he likes the result?

**Comment by muramura** [comment]
I verified this PR with PIXHAWK1 and FUTABA propo.
1. "RC NOT FOUND" is displayed when the radio is turned off.
2. no message is notified when the radio is turned on.
3. Fail-safe is recognized when the radio is turned off.

I had satisfied 1-3.

Thanks.

THIS PR
Screenshot from 2022-10-09 11-55-36

File: libraries/RC_Channel/RC_Channel.h [hunk 1]
```diff
@@ -614,6 +618,7 @@ class RC_Channels {
     uint32_t last_update_ms;
     bool has_new_overrides;
     bool _has_had_rc_receiver; // true if we have had a direct detach RC reciever, does not include overrides
+    bool _had_had_override; // true if we have had a override on any channel
```
Comment (rmackay9) [inline libraries/RC_Channel/RC_Channel.h:621]: sorry to be annoying but this should probably be, "_ha**s**_had_override"?

File: libraries/RC_Channel/RC_Channel.h [hunk 2]
```diff
@@ -559,6 +559,9 @@ class RC_Channels {
     // returns true if we have had a direct detach RC reciever, does not include overrides
     bool has_had_rc_receiver() const { return _has_had_rc_receiver; }
 
+    // returns true if we have had a overrides on any channel
```
Comment (rmackay9) [inline libraries/RC_Channel/RC_Channel.h:562]: I guess this comment should be "if we have had an override on any channel"?

**Comment by rmackay9** [comment]
@IamPete1, txs again for this.  I flight tested this today in a competition and all went well.

Feel free to merge before or after you make the spelling fixes. :-)

File: libraries/RC_Channel/RC_Channel.h (hunk 1 above)
Comment (IamPete1) [reply libraries/RC_Channel/RC_Channel.h]: fixed

File: libraries/RC_Channel/RC_Channel.h (hunk 2 above)
Comment (IamPete1) [reply libraries/RC_Channel/RC_Channel.h]: fixed

**Comment by WickedShell** [comment]
At this point why do we need another flag? Why not just check `last_input_ms()` is not zero?

This code is almost present already in `AP_Arming::rc_arm_checks()` I'd actually argue that we should remove the special case of `last_input_ms() == 0` and just make that a failure in it's own right, and factor it to the common code side.

**Pull request merged**